frontends/parsers/p4/p4parser.ypp

```diff
@@ -440,8 +441,14 @@ optAnnotations
     ;
 
 annotations
-    : annotation  { $$ = new IR::Vector<IR::Annotation>(); $$->push_back($1); }
-    | annotations annotation { $$ = $1; $$->push_back($2); }
+    : annotation  {
+       $$ = new IR::Vector<IR::Annotation>();
+       if (! P4CContext::get().options().isAnnotationDisabled($1))
```

Comment:
wouldn't it be somewhat cleaner to write a pass to remove the annotations rather than not insert them?

Reply:
Why would that be cleaner? Especially if you want to ensure that only the user defined annotations are removed and the compiler can insert such annotations for its used, by not adding the annotations in the parser you avoid issues with pass ordering.

Reply:
It's cleaner because the parser just parses and does not look at options.
Also, a visitor for this would be very simple indeed: just return null for the annotation that matches the name.